File: Examples/CSharp/AsposePdfFacades/Bookmarks/ExtractBookmarks.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Facades;
using System;

namespace CSharp.AsposePDFFacades.Bookmarks
{
    public class ExtractBookmarks
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdfFacades_Bookmarks();
            //create PdfBookmarkEditor
            PdfBookmarkEditor bookmarkEditor = new PdfBookmarkEditor();
            //open PDF file
            bookmarkEditor.BindPdf(dataDir+ "ExtractBookmarks.pdf");
            //extract bookmarks
            Aspose.Pdf.Facades.Bookmarks bookmarks = bookmarkEditor.ExtractBookmarks();

            foreach (Bookmark bookmark in bookmarks)
            {
                Console.WriteLine("Title: {0}", bookmark.Title);
                Console.WriteLine("Page Number: {0}", bookmark.PageNumber);
            }
 
            
            
        }
    }
}

File: Examples/VisualBasic/AsposePdfFacades/Bookmarks/ExtractBookmarks.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Facades

Namespace AsposePDFFacades.Bookmarks
    Public Class ExtractBookmarks
        Public Shared Sub Run()
            ' ExStart:ExtractBookmarks
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdfFacades_Bookmarks()
            ' Create PdfBookmarkEditor
            Dim bookmarkEditor As New PdfBookmarkEditor()
            ' Open PDF file
            bookmarkEditor.BindPdf(dataDir & Convert.ToString("ExtractBookmarks.pdf"))
            ' Extract bookmarks
            Dim bookmarks As Aspose.Pdf.Facades.Bookmarks = bookmarkEditor.ExtractBookmarks()

            For Each bookmark As Bookmark In bookmarks
                Console.WriteLine("Title: {0}", bookmark.Title)
                Console.WriteLine("Page Number: {0}", bookmark.PageNumber)
            Next
            ' ExEnd:ExtractBookmarks
        End Sub
    End Class
End Namespace
